src/lazy/expanded/template.rs

```diff
@@ -994,32 +971,20 @@ impl<'top> TemplateMacroInvocation<'top> {
             self.context,
             environment,
             self.arg_expressions(),
-            self.host_macro_ref(),
+            self.host_template(),
         )
     }
-    pub fn host_template_address(&self) -> MacroAddress {
-        self.host_template_address
-    }
 
     /// Helper method to access the definition of the host template. Useful for debugging,
     /// but not required for macro expansion.
-    pub fn host_macro_ref(&self) -> MacroRef<'top> {
-        self.context()
-            .macro_table()
-            .macro_at_address(self.host_template_address)
-            .unwrap()
+    pub fn host_macro_ref(&self) -> &'top Macro {
+        self.host_template.macro_ref
```

Comment:
🗺️ Here and below: more address lookups and unwrapping obviated by holding an actual reference.